sql/parser/builtins.go

```diff
@@ -382,8 +382,8 @@ var builtins = map[string][]builtin{
 			types:      typeList{timestampType},
 			returnType: DummyInterval,
 			impure:     true,
```

Comment:
I've updated the comment to reflect some of these thoughts. I prefer not mentioning how impure is used and the comments reflect only how a builtin writer would specify it.